File: Project_Euler/Python/p1.py

import time

a = 0
z = 0

target = input("Sum all multiples 3 and 5 below: ")

start = time.time()

while a<target-1:
    a += 1
    if a % 3 == 0 or a % 5 == 0:
        z += a

elapsed = (time.time() - start)

print("Implementation 1 found %s in %s seconds " % (z, elapsed))


print("\n")



###New implementantion below
start = time.time()

result = int(sum(list(i for i in range(1, target) if i%3==0 or i%5==0)))

elapsed = (time.time() - start)

print("Implementation 2 found %s in %s seconds " % (result, elapsed))

File: Project_Euler/CPP/p1.cpp
//Sums all multiples of 3 and 5 below "target"
//My firsrt ever C++ <3

#include<iostream>
#include<string>
#include<ctime>
using namespace std;

int main() {
  int count;
  long int target;
  long int sum;
  std::cout << "Enter target: \n";
  cin >> target;
  while (count < target-1) {
    count += 1;
    if (count % 3 == 0 || count % 5 == 0) {
      sum += count;
    }
  }
  std::cout << "The sum of all multiples of 3 and 5 below " << target << " is " << sum << "\n" << endl;
}
